src/main/java/com/google/cloud/teleport/splunk/SplunkEventWriter.java

```diff
@@ -225,24 +230,32 @@ private void flush(
             e.getStatusCode(), e.getContent(), e.getStatusMessage());
         logWriteFailures(countState);
 
-        String payload = publisher.getStringPayload(events);
-        SplunkWriteError error = SplunkWriteError.newBuilder()
-            .withStatusCode(e.getStatusCode())
-            .withStatusMessage(e.getStatusMessage())
-            .withPayload(payload)
-            .build();
-        receiver.output(error);
+        for (SplunkEvent event : events) {
+          String payload = GSON.toJson(event);
+          SplunkWriteError error =
+              SplunkWriteError.newBuilder()
+                  .withStatusCode(e.getStatusCode())
+                  .withStatusMessage(e.getStatusMessage())
+                  .withPayload(payload)
+                  .build();
+          
+          receiver.output(error);
+        }
```

Comment:
Lets see what we can do here to avoid duplicating this block twice in both the exception cases.

Reply:
Shifted these blocks to a new method `flushWriteFailures`.